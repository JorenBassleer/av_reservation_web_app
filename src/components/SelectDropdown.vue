<template>
  <select
    v-model="rowId"
    @change="clickedOnRow(rowId)"
  >
    <option
      v-for="row in data"
      :key="row.id"
      :value="row.id"
    >
      {{ row.name }}
    </option>
  </select>
</template>
<script setup>
import { ref, defineProps, defineEmits } from 'vue';

const props = defineProps({
  data: {
    type: [Object, Array],
    required: true,
  },
  type: {
    type: [Object, String],
    required: true,
  },
});
const emit = defineEmits(['clickedOnRow']);
const rowId = ref(0);
const clickedOnRow = (clickedId) => {
  emit('clickedOnRow', {
    clickedId,
    type: props.type,
  });
};
</script>
<style scoped>
select, option {
    border: none;
    outline: inherit;
    padding: 10px;
    width: 280px;
    border-radius: 8px;
    text-align: center;
}
</style>
