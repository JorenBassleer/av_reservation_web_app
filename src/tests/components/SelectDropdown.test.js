import { shallowMount } from '@vue/test-utils';
import {
  expect, describe, it, beforeEach,
} from 'vitest';
import SelectDropdown from '../../components/SelectDropdown.vue';

describe('SelectDropdown.vue', () => {
  let wrapper;
  beforeEach(() => {
    // Arrange
    wrapper = shallowMount(SelectDropdown, {
      props: {
        data: [],
        type: 'typeName',
      },
    });
  });
  it('Match snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
  it('Emit row id with prop.type', () => {
    // Act
    wrapper.vm.clickedOnRow(1);
    wrapper.vm.$nextTick(() => {
    // Assert
      expect(wrapper.emitted().clickedOnRow[0]).toEqual([
        {
          rowId: 1,
          type: 'typeName',
        },
      ]);
    });
  });
});
