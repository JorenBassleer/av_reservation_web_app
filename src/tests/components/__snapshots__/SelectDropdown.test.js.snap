// Vitest Snapshot v1

exports[`SelectDropdown.vue > Match snapshot 1`] = `"<select data-v-0484f41b=\\"\\"></select>"`;
